From 98df47b94deeee784e36f6084b2ac8c18b4d51f1 Mon Sep 17 00:00:00 2001
From: jiangxin <jiangxin@hygon.cn>
Date: Wed, 25 Aug 2021 12:36:00 +0800
Subject: [PATCH 7/9] anolis: csv/i386: CSV guest do not need
 register/unregister guest secure memory

CSV guest memory is allocated by firmware in secure processor
from dedicated memory reserved upon system boot up,
consequently it is not necessary to add notifier to pin/unpin memory.

Signed-off-by: Xin Jiang <jiangxin@hygon.cn>
Change-Id: Idfe109b3925971ed71116bed2975133255ba11cc
---
 target/i386/sev.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/target/i386/sev.c b/target/i386/sev.c
index efbe48fc30..6a672c4398 100644
--- a/target/i386/sev.c
+++ b/target/i386/sev.c
@@ -1013,7 +1013,10 @@ int sev_kvm_init(ConfidentialGuestSupport *cgs, Error **errp)
         goto err;
     }
 
-    ram_block_notifier_add(&sev_ram_notifier);
+    /* CSV guest needs no notifier to reg/unreg memory */
+    if (!csv_enabled()) {
+        ram_block_notifier_add(&sev_ram_notifier);
+    }
     qemu_add_machine_init_done_notifier(&sev_machine_done_notify);
     qemu_add_vm_change_state_handler(sev_vm_state_change, sev);
 
-- 
2.17.1

